### PR TITLE
Fixes #1665 use old compaction thread pool config if set

### LINKED ISSUE #1665: Preserve user intent with regard to overridden compaction thread pool size
As mentioned on #1662, the new property `tserver.compaction.major.service.default.planner.opts.executors` overrides any user-specified value in the now deprecated `tserver.compaction.major.concurrent.max` property.

My proposed solution is:

* Change the default value of the old property to: `0`
* If the old property is not configured to its default value, then it must be overridden by the user, in which case, we should:
  * Infer the user wishes a single thread pool like: `[{"name":"deprecation-inferred-single-pool","numThreads":"X"}]`, where `X` is the value configured in the old property, and
  * Issue a warning that the old property is deprecated and to show the inferred value for the new property.
* If both properties' default values are overridden, then we should:
  * Use the new property's value, and
  * Issue a warning about the old one being set and show the new property's value that will be used.
* If only the new property, or neither, or overridden from their default values, then the new value will be used with no warnings.

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

**Comment by keith-turner** [comment]
I tested this change manually using Uno.  In one test I set only the the old deprecated property and inspected the logs to see that the expected warning was logged and the an executor w/ the expected number of threads was created.  In another test I set the old deprecated prop and also modified the new prop.  For that test I could see in the logs that it logged a warning, but did not use the old prop.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/compactions/CompactionManager.java [hunk 1]
```diff
@@ -78,10 +81,82 @@ private static long getDefaultThroughput(AccumuloConfiguration aconf) {
...
+
+          Map<String,String> defaultServiceConfigs =
+              Map.of(defaultServicePrefix + "planner", DefaultCompactionPlanner.class.getName(),
+                  defaultServicePrefix + "planner.opts.executors",
+                  "[{'name':'deprecated','numThreads':" + numThreads + "}]");
+
+          configsCopy.putAll(defaultServiceConfigs);
+
+          String warning = String.format(
+              "The deprecated property %s was set.  Properties with the prefix %s "
```
Comment (ctubbsii) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/compactions/CompactionManager.java:137]: You have some inconsistency, sometimes using double spaces after a period, and sometimes using a space after a period. Although there is debate over which is best, I recommend standardizing on a single space after the period. It is more concise, and easier to predict and grep for in the logs. When they show on the monitor page and presented as HTML, they get collapsed to a single space anyway, so matching in a browser with Ctrl-F is also consistent with grepping for a single space in the logs.

**Review by ctubbsii — COMMENTED** [review]
Looks okay to me. I have only minor / trivial suggestions.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/compactions/CompactionManager.java [hunk 2]
```diff
@@ -78,10 +81,82 @@ private static long getDefaultThroughput(AccumuloConfiguration aconf) {
...
+        // check if any properties for the default compaction service are set
+        for (String key : configs.keySet()) {
+          if (key.startsWith(defaultServicePrefix)) {
+            Property prop = Property.getPropertyByKey(key);
+            if (prop == null || aconf.isPropertySet(prop, true)) {
+              defaultServicePropsSet = true;
+              break;
+            }
+          }
+        }
```
[the remark this answers is not in the document]
Comment (keith-turner) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/compactions/CompactionManager.java]: I made that change and it worked out quite nicely.  Before merging I will manually test again to make sure I didn't mess anything up replacing the loop with a stream.

**Review by ctubbsii — APPROVED** [review]

**Comment by keith-turner** [comment]
I manually retested after making code review changes and everything worked as expected.

**Pull request merged**